assembly/src/assembler/mod.rs

```diff
@@ -67,7 +67,16 @@ impl Assembler {
         L: Library,
         I: Iterator<Item = L>,
     {
-        libraries.try_fold(self, |slf, library| slf.with_library(&library))
+        let assembler = libraries.try_fold(self, |slf, library| slf.with_library(&library));
+        #[cfg(debug_assertions)]
+        {
+            let libraries_vec = libraries.collect::<Vec<L>>();
+            assert!(libraries_vec.iter().all(|lib| lib
+                .dependencies()
+                .iter()
+                .all(|dep| libraries_vec.iter().any(|l| l.root_ns() == dep))));
+        }
```

Comment:
Since we currently don't impose any order in which libraries could be loaded, this theoretically could lead to a failure when it shouldn't. For example, the user could use this method to load some libraries, but then use `with_library()` method to load a library which is a dependency for one of these libraries.

I do think this would be a great check to have, but I think we need to put in more thinking into how it would work.